GRDB/Record/PersistableRecord.swift

```diff
@@ -841,8 +871,19 @@ extension UpdateQuery {
         if let sql = Self.sqlCache[self] {
             return sql
         }
+        let sql = uncachedSQL()
+        Self.sqlCache[self] = sql
+        return sql
+    }
+    
+    func uncachedSQL() -> String {
         let updateSQL = updatedColumns.map { "\($0.quotedDatabaseIdentifier)=?" }.joined(separator: ", ")
-        let whereSQL = conditionColumns.map { "\($0.quotedDatabaseIdentifier)=?" }.joined(separator: " AND ")
+        var whereSQL = conditionColumns.map { "\($0.quotedDatabaseIdentifier)=?" }.joined(separator: " AND ")
+        if nullConditionColumns.isEmpty == false {
+            let nullConditions = nullConditionColumns.map { "\($0.quotedDatabaseIdentifier) IS NULL" }
+            whereSQL.append("AND \(nullConditions.joined(separator: " AND "))")
```

Comment:
I agree with the code suggestion and will make it so! But to note, currently conditionColumns is never empty because I retained the condition where the query returns early if all the primary keys are nil / null: https://github.com/groue/GRDB.swift/blob/97221c7bd63fbd82dd7303faf202e956ef268722/GRDB/Record/PersistableRecord.swift#L698-L700

Reply:
Fixed in 3e05dae